### PR TITLE
Update `platform::rawfile`

### PR DESCRIPTION
**Pull request by stratact**

### CODE REVIEW

**Comment by MggMuggins** [comment]
I'd approve this, but I'm not sure if anything needs to be done to make it cross-platform.

File: src/platform/src/rawfile.rs [hunk 1]
```diff
@@ -1,20 +1,39 @@
 use core::ops::Deref;
+use sys::{open, dup, close};
```
Comment (MggMuggins) [inline src/platform/src/rawfile.rs]: Isn't there a `pub use sys::*` in `lib.rs`?

Comment (stratact) [reply src/platform/src/rawfile.rs]: Actually, it turns out there is. https://github.com/redox-os/relibc/blob/master/src/platform/src/lib.rs#L15

So, should I do `platform::{open, dup, close};` instead?

File: src/platform/src/rawfile.rs [hunk 2]
```diff
@@ -1,20 +1,39 @@
 use core::ops::Deref;
+use super::{open, dup, close};
 
 pub struct RawFile(usize);
 
 impl RawFile {
-    pub fn open<T: AsRef<[u8]>>(path: T, flags: usize) -> Result<RawFile> {
-        open(path, flags).map(RawFile)
+    pub fn open<T: AsRef<[u8]>>(path: T, flags: usize, mode: usize) -> Result<RawFile, ()> {
+        match open(path.as_ref()[0] as *const i8, flags as i32, mode as u16) {
```
[the remark this answers is not in the document]
Comment (stratact) [reply src/platform/src/rawfile.rs]: So should I make `path` into a `Vec<u8>` and push a `'\0'` character into it to make it zero-terminated?

[the remark this answers is not in the document]
Comment (stratact) [reply src/platform/src/rawfile.rs]: I guess doing a Vec would be too expensive though...

File: src/platform/src/rawfile.rs [hunk 3]
```diff
@@ -1,32 +1,37 @@
 use core::ops::Deref;
-use super::{open, dup, close};
+use super::{open, dup, close, types::*};
 
-pub struct RawFile(usize);
+trait Into<c_char> {
```
Comment (jackpot51) [inline src/platform/src/rawfile.rs]: Is this required anymore?

**Pull request merged**